src/app/modules/verification/verification.routes.ts

```diff
@@ -1,65 +1,74 @@
 import { celebrate, Joi } from 'celebrate'
-import { Express } from 'express'
+import { Router } from 'express'
 
 import verifiedFieldsFactory from './verification.factory'
 
-const mountVfnRoutes = (app: Express): void => {
-  const formatOfId = Joi.string().length(24).hex().required()
-  app.route('/transaction').post(
-    celebrate({
-      body: Joi.object({
-        formId: formatOfId,
-      }),
+const VfnRouter = Router()
+
+const formatOfId = Joi.string().length(24).hex().required()
+
+VfnRouter.post(
+  '/',
+  celebrate({
+    body: Joi.object({
+      formId: formatOfId,
+    }),
+  }),
+  verifiedFieldsFactory.createTransaction,
+)
+
+VfnRouter.get(
+  '/:transactionId',
+  celebrate({
+    params: Joi.object({
+      transactionId: formatOfId,
     }),
-    verifiedFieldsFactory.createTransaction,
-  )
-  app.route('/transaction/:transactionId').get(
-    celebrate({
-      params: Joi.object({
-        transactionId: formatOfId,
-      }),
+  }),
+  verifiedFieldsFactory.getTransactionMetadata,
+)
+
+VfnRouter.post(
+  '/:transactionId/reset',
+  celebrate({
+    params: Joi.object({
+      transactionId: formatOfId,
+    }),
+    body: Joi.object({
+      fieldId: formatOfId,
     }),
-    verifiedFieldsFactory.getTransactionMetadata,
-  )
-  app.route('/transaction/:transactionId/reset').post(
-    celebrate({
-      params: Joi.object({
-        transactionId: formatOfId,
-      }),
-      body: Joi.object({
-        fieldId: formatOfId,
-      }),
+  }),
+  verifiedFieldsFactory.resetFieldInTransaction,
+)
+
+VfnRouter.post(
+  '/:transactionId/otp',
+  celebrate({
+    params: Joi.object({
+      transactionId: formatOfId,
     }),
-    verifiedFieldsFactory.resetFieldInTransaction,
-  )
-  app.route('/transaction/:transactionId/otp').post(
-    celebrate({
-      params: Joi.object({
-        transactionId: formatOfId,
-      }),
-      body: Joi.object({
-        fieldId: formatOfId,
-        answer: Joi.string().required(),
-      }),
+    body: Joi.object({
+      fieldId: formatOfId,
+      answer: Joi.string().required(),
     }),
-    verifiedFieldsFactory.getNewOtp,
-  )
+  }),
+  verifiedFieldsFactory.getNewOtp,
+)
 
-  app.route('/transaction/:transactionId/otp/verify').post(
-    celebrate({
-      params: Joi.object({
-        transactionId: formatOfId,
-      }),
-      body: Joi.object({
-        fieldId: formatOfId,
-        otp: Joi.string()
-          .regex(/^\d{6}$/)
-          .required()
-          .error(() => 'Please enter a valid otp'),
-      }),
+VfnRouter.post(
+  '/:transactionId/otp/verify',
+  celebrate({
+    params: Joi.object({
+      transactionId: formatOfId,
+    }),
+    body: Joi.object({
+      fieldId: formatOfId,
+      otp: Joi.string()
+        .regex(/^\d{6}$/)
+        .required()
+        .error(() => 'Please enter a valid otp'),
```

Comment:
`error(() => 'xxx')` has been used wrongly all this time, use `.message()` instead

```suggestion
        .message('Please enter a valid otp'),
```

See [documentation](https://joi.dev/api/?v=17.2.1#anyerrorerr) where it redirects us to use [`.message`](https://joi.dev/api/?v=17.2.1#anymessagemessage)

Reply:
hmm, this broke joi validation. https://github.com/sideway/joi/issues/2213

Reply:
Really? I used it on AuthController PR (WIP) without issues. How did it break joi validation?